src/journal.c

```diff
@@ -333,561 +332,540 @@ static void sub_0202B898(Journal *journal, UnkStruct_0202BFCC *param1)
 
     v0->unk_00 = 16;
     return v0;
 }
 
-void *sub_0202BDE0(u32 param0)
+void *sub_0202BDE0(u32 heapID)
 {
-    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(param0);
+    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(heapID);
 
     v0->unk_00 = 17;
     return v0;
 }
 
-void *sub_0202BDEC(u16 param0, u32 param1)
+void *sub_0202BDEC(u16 param0, u32 heapID)
 {
-    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(param1);
+    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(heapID);
 
     v0->unk_00 = 18;
     v0->unk_06 = param0;
 
     return v0;
 }
 
-void *sub_0202BE00(u8 param0, u16 param1, u32 param2)
+void *sub_0202BE00(u8 param0, u16 param1, u32 heapID)
 {
-    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(param2);
+    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(heapID);
 
     v0->unk_00 = 19 + param0;
     v0->unk_02 = param1;
 
     return v0;
 }
 
-void *sub_0202BE14(u32 param0)
+void *sub_0202BE14(u32 heapID)
 {
-    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(param0);
+    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(heapID);
 
     v0->unk_00 = 34;
     return v0;
 }
 
-void *sub_0202BE20(u32 param0)
+void *sub_0202BE20(u32 heapID)
 {
-    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(param0);
+    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(heapID);
 
     v0->unk_00 = 35;
     return v0;
 }
 
-void *sub_0202BE2C(u32 param0, u32 param1)
+void *sub_0202BE2C(u32 heapID, u32 param1)
 {
-    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(param0);
+    UnkStruct_0202BCC8 *v0 = sub_0202BCC8(heapID);
 
     v0->unk_00 = param1;
     return v0;
 }
 
-static UnkStruct_0202BE38 *sub_0202BE38(u32 param0)
+static JournalEntryMon *JournalEntry_InitMon(u32 heapID)
 {
-    UnkStruct_0202BE38 *v0 = Heap_AllocFromHeap(param0, sizeof(UnkStruct_0202BE38));
+    JournalEntryMon *journalEntryMon = Heap_AllocFromHeap(heapID, sizeof(JournalEntryMon));
 
-    memset(v0, 0, sizeof(UnkStruct_0202BE38));
-    return v0;
+    memset(journalEntryMon, 0, sizeof(JournalEntryMon));
+    return journalEntryMon;
 }
 
-void *sub_0202BE4C(const PlayTime *playTime, u16 param1, u8 param2, u8 param3, u32 param4)
+void *JournalEntry_CreateMonCaught(const PlayTime *playTime, u16 species, u8 gender, u8 timeOfDay, u32 heapID)
 {
-    UnkStruct_0202BE38 *v0;
+    JournalEntryMon *journalEntryMon = JournalEntry_InitMon(heapID);
     u8 v1;
 
-    v0 = sub_0202BE38(param4);
-
-    v0->unk_00 = 1;
-    v0->unk_02 = param1;
-    v0->unk_01_6 = param2;
-    v0->unk_01_2 = param3;
+    journalEntryMon->battleResult = POKEMON_CAUGHT;
+    journalEntryMon->species = species;
+    journalEntryMon->gender = gender;
+    journalEntryMon->timeOfDay = timeOfDay;
 
     v1 = PlayTime_GetMinutes(playTime) / 10;
 
     if ((v1 == 0) || (v1 == 2) || (v1 == 4)) {
-        v0->unk_01_0 = 0;
+        journalEntryMon->stringVariant = 0;
     } else if ((v1 == 1) || (v1 == 3)) {
-        v0->unk_01_0 = 1;
+        journalEntryMon->stringVariant = 1;
     } else {
-        v0->unk_01_0 = 2;
+        journalEntryMon->stringVariant = 2;
```

Comment:
**question**: Should these use the `define`s specified in `journal.h`?

Reply:
No, with `stringVariant` I just mean slight variations of basically the same string, not the difference between catching and defeating a pokemon. This is for example from the function that prints the strings for caught mons:
```    
switch (journalEntryMon->stringVariant) {
    case 0:
        v0 = MessageLoader_GetNewStrbuf(param0->unk_50, 49); // Caught [species] ([time]).
        break;
    case 1:
        v0 = MessageLoader_GetNewStrbuf(param0->unk_50, 50); // [species] was caught ([time]).
        break;
    default:
        if (journalEntryMon->gender == 0) {
            v0 = MessageLoader_GetNewStrbuf(param0->unk_50, 51); // Caught a male [species].
        } else if (journalEntryMon->gender == 1) {
            v0 = MessageLoader_GetNewStrbuf(param0->unk_50, 52); // Caught a female [species].
        } else {
            v0 = MessageLoader_GetNewStrbuf(param0->unk_50, 49); // Caught [species] ([time]).
        }
    }
```
You can see that it just determines some slight variation in word order to keep the journal more interesting, I don't think there's a need to use any `define`s for that.

Reply:
Interesting.

Now that I notice then, though, could you add constants from the appropriate text bank? Judging from [this entry](https://github.com/pret/pokeplatinum/blob/f172250fae8c3e00285f746cde9f4e29de9de6b6/res/text/message_bank_unk_0366.gmm#L210), it looks like the include should be `gmm/pl_msg_00000366.h`, but it would be great if you could also:

1. Rename the `gmm` file to give it a meaningful name (maybe `message_bank_journal_entries.gmm`) -- make sure to also update `res/text/meson.build` and `res/text/message_banks.order`.
2. Update the entries in the file itself to use an appropriate prefix instead of `pl_msg_00000366` (maybe just `journal_entries`).